### PR TITLE
feat(query): support `GROUP BY GROUPING SETS` clause.

### PR DESCRIPTION
**Pull request by RinChanNOWWW**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Basic implementation of `GROUP BY GROUPING SETS`.

Main idea:

Add a new `PhysicalPlan`: `PhysicalPlan::AggregateExpand` and its corresponding `Transform`: `TransformExpandGroupingSets`.

If the query is a `GROUP BY GROUPING SETS` clause, `PhysicalPlan::AggregateExpand` will be added to the front of `PhysicalPlan::AggregatePartial`. And the plan will be built into `TransformExpandGroupingSets`.

`TransformExpandGroupingSets` will expand the input `DataBlock` for each grouping set. It will fill NULLs if the group-by columns are not found in the grouping set.

Tacking in #10492

### CODE REVIEW

**Comment by sundy-li** [comment]
> TransformExpandGroupingSets will expand the input DataBlock for each grouping set. It will fill NULLs if the group-by columns are not found in the grouping set.

What if the original column has nulls? Seem they share the same hashmap here.

**Review by sundy-li — APPROVED** [review]
Rest Looks good to me.

File: src/query/sql/src/planner/binder/aggregate.rs [hunk 1]
```diff
@@ -74,6 +78,11 @@ pub struct AggregateInfo {
     /// TODO(leiysky): so far we are using `Debug` string of `Scalar` as identifier,
     /// maybe a more reasonable way is needed
     pub group_items_map: HashMap<String, usize>,
+
+    /// Index for virtual column `grouping_id`. It's valid only if `grouping_sets` is not empty.
+    pub grouping_id_index: IndexType,
```
Comment (xudong963) [inline src/query/sql/src/planner/binder/aggregate.rs:83]: Better to be `Option<IndexType>`? If `grouping_sets` is empty, it's `None`

Comment (RinChanNOWWW) [reply src/query/sql/src/planner/binder/aggregate.rs:83]: Option<IndexType> will consume more memory.

image

**Review by xudong963 — APPROVED** [review]

**Comment by RinChanNOWWW** [comment]
image

cc @BohuTANG

**Pull request merged**